src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -1859,23 +1870,10 @@ public override void List(CriteriaImpl criteria, IList results)
 
 				string[] implementors = Factory.GetImplementors(criteria.EntityOrClassName);
 				int size = implementors.Length;
-
```

Comment:
I think this is a [failed] attempt to remove double persister resolution here. `GetImplementors` gets the persister from by a class name and then returns `EntityName`, and `GetOuterJoinLoadable` gets the persister again.